src/main/java/org/isf/admission/manager/AdmissionBrowserManager.java

```diff
@@ -502,6 +504,7 @@ else if (dateOut.isBefore(ad.getAdmDate()) || dateOut.equals(ad.getAdmDate())) {
 				// date control
 				if (admission.getDeliveryDate() == null) {
 					errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.controln1datenodeliverydatefound.msg")));
+					throw new OHDataValidationException(errors);
```

Comment:
Why in this case we throw the error directly?

Reply:
Because the persistence code in the following block of code requires them to be non-null.

<pre>
		/*
		 * Initialize AdmissionBrowserManager
		 */
		Patient patient = admission.getPatient();
		List<Admission> admList = getAdmissions(patient);
</pre>

Reply:
Sorry, I don't understand this.

Also in lines 314 and 320 should not be there the `throw new`, unless we want to stop immediately the validation at that point (makes sense).

Now that I think about it, it seems we miss a check for patient == null, is it?

Reply:
ok, got it, but so we could put it in a `else` block?

same for the following check, where `admission.getCtrlDate1() == null` could be replaced with `ctrl1Date`:

```
			LocalDateTime ctrl2Date = admission.getCtrlDate2();
			if (ctrl2Date != null) {
				if (admission.getCtrlDate1() == null) {
					errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.controldaten2controldaten1notfound.msg")));
				}
				// date control
				LocalDateTime limit;
				if (admission.getDisDate() == null) {
					limit = today;
				} else {
					limit = admission.getDisDate();
				}
				if (ctrl1Date != null && (ctrl2Date.isBefore(ctrl1Date) || ctrl2Date.isAfter(limit))) {
					errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.pleaseinsertavalidcontroln2date.msg")));
				}
			}
```

Reply:
I have this change ready once other issues are decided on.

Reply:
committed

Reply:
My suggestion here is not to throw the exception on line 500 but rather using an `else` in line 501.
Same at line 517, where since the exception is not thrown the `ctrl1Date` could throw NPE in line 525

Reply:
done